driver-core/src/main/com/mongodb/client/model/TextSearchOptions.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model;
+
+/**
+ * Text search options for the {@link Filters#text(String, TextSearchOptions)} helper
+ *
+ * @mongodb.driver.manual reference/operator/query/text $text
+ * @since 3.2
+ */
+public class TextSearchOptions {
+
+    private String language;
+    private Boolean caseSensitive;
```

Comment:
Currently if the user explicitly sets it, then we explicitly set it, to me this is also future proof, you could easily envisage case sensitivity becoming an index option for creation of the index.